File: Kotlin/src/main/kotlin/d_while_loops/exercise_46.kt
package d_while_loops

// 46) Create a program that calculates and displays the result of the sum between 6 + 8 + 10 + 12 + 14 + … + 98 + 100.

fun main() {
    var counter = 6
    var sum = 0

    while (counter <= 100) {
        sum += counter
        counter += 2
    }
    print("Sum: $sum")
}

File: Dart/d_while_loops/exercise_46.dart
// 46) Create a program that calculates and displays the result of the sum between 6 + 8 + 10 + 12 + 14 + … + 98 + 100.

void main() {
  int startNumber = 6;

  int sum = calculateSum(startNumber);
  print('Sum: $sum');
}

int calculateSum(int number) {
  int sum = 0;
  int maxNumber = 100;
  int increment = 2;

  while (number <= maxNumber) {
    sum += number;
    number += increment;
  }

  return sum;
}
